chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -79,6 +79,42 @@ object Module {
   def currentModule: Option[BaseModule] = Builder.currentModule
 }
 
+private[chisel3] trait IOImpl {
```

Comment:
Does this need to be a trait, or can this just be an object? It doesn't appear you're extending the trait anywhere, and you should be able to forward vals in the package object?

Reply:
The trait is extended by a private object IO used internally to chisel3.core and by an object in package chisel3.experimental (in package.scala). I was kind of prototyping a new way of doing the val forwarding where the in the ScalaDoc the actual documentation exists where you expect it to instead of having to go look in chisel3.core. I grant that perhaps this PR isn't the appropriate place to try out changing how the documentation exists in the ScalaDoc so I'll switch to the existing pattern.